packages/grafana-ui/src/components/ClipboardButton/ClipboardButton.tsx

```diff
@@ -1,51 +1,26 @@
-import React, { PureComponent } from 'react';
-import Clipboard from 'clipboard';
+import React, { useCallback } from 'react';
 import { Button, ButtonProps } from '../Button';
 
 export interface Props extends ButtonProps {
   /** A function that returns text to be copied */
   getText(): string;
   /** Callback when the text has been successfully copied */
-  onClipboardCopy?(e: Clipboard.Event): void;
+  onClipboardCopy?(): void;
```

Comment:
Would it be possible to keep backward compatibility by introducing our own ClipboardEvent type that looks the same as the `Clipboard.Event`? Or is that a complex type to manage?

Reply:
I'll look into it

Reply:
So the interface is like this:
```ts
    interface Event {
        action: string;
        text: string;
        trigger: Element;
        clearSelection(): void;
    }
```

The trouble is some of these don't really make sense (e.g. `clearSelection`) now that the Clipboard API is being used, so I'm not sure what the best move is.

Reply:
Ok ok, I will see if I can find any stats on how widely this is used.

Reply:
From my looks within grafana, no one uses it. Can't comment for _outside_ of grafana.,

I would consider providing a "dummy" event object, at least to prevent any `cannot read property action of undefined` exceptions.

Reply:
Not the most scientific test, but I downloaded every plugin supported by the current version of Grafana and grepped for `ClipboardButton` with no results, which makes me think it's not used by anyone. But then again, a breaking change is a breaking change

Reply:
I think you can carry on with this change. I have checked our collected data and can't find anyone using this. However, I have only checked the react code. So there might be usages of the angular directive.